Modules/_collectionsmodule.c

```diff
@@ -1848,6 +2015,7 @@ dequereviter_next(dequeiterobject *it)
         it->b = it->b->leftlink;
         it->index = BLOCKLEN - 1;
     }
+    deque_unlock(it->deque);
```

Comment:
We don't need a separate lock for the iterator because we can use the deque's lock (since the iterator doesn't mutate the deque), right?  Wouldn't a separate lock for the iterator be more efficient?